test.py

```diff
@@ -424,8 +442,18 @@ def build_test_list(self) -> List[str]:
     def pattern(self) -> str:
         assert False
 
-    async def add_test(self, shortname) -> None:
-        test = PythonTest(self.next_id((shortname, self.suite_key)), shortname, self)
+    def cases(self, shortname: str) -> List[str]:
+        # Find test cases in pytest file (without importing), so they can be invoked separately
```

Comment:
Not just for parallel, we it can do partial name match.

Also I had problems using quoted calls with `pytest -k 'blah'`, so better to call them separately, IMHO.